lib/flash_flow/github_lock.rb

```diff
@@ -0,0 +1,41 @@
+require 'action_view'
+require 'action_view/helpers'
+include ActionView::Helpers::DateHelper
+require 'flash_flow/github'
+require 'flash_flow/lock'
+
+module FlashFlow
+  class GithubLock < Lock
+    attr_accessor :repo
+
+    def initialize(repo)
+      @github = Github.new(repo)
+    end
+
+    def with_lock(issue_id, &block)
+      return block.call if issue_id.nil?
+
+      if @github.issue_open?(issue_id)
+        last_event = @github.get_last_event(issue_id)
+        actor = last_event[:actor][:login]
+        time = time_ago_in_words(last_event[:created_at])
```

Comment:
Also, I would move all of this code into the error_message method, since it's only used for the error message. It will make the `with_lock` method easier to understand.